connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkConnectClient.scala

```diff
@@ -60,14 +63,40 @@ private[sql] class SparkConnectClient(
     new ArtifactManager(userContext, sessionId, channel)
   }
 
+  private val retryParameters: SparkConnectClient.RetryParameters = configuration.retryParameters
+
+  @tailrec private[client] final def retry[T](fn: => T, retries: Int = 0): T = {
+    if (retries > retryParameters.max_retries) {
+      throw new IllegalArgumentException(s"retries must not exceed retryParameters.max_retries")
+    }
+    Try {
+      fn
```

Comment:
Similar to [Python Client](https://github.com/databricks/runtime/blob/5d6d813dbf9d993df45fdb459eb8b6a6d647324e/python/pyspark/sql/connect/client/core.py#L559-L584), we should implement the logic that determines whether an RuntimeStatusException is retryable based the status code?

Reply:
I chose to retry on no exception by default to not alter the behavior, but I agree, mirroring the Python client here makes more sense!

Reply:
Added a default retryException similar to the Python client + adjusted the tests